src/Network/NWAdvertiseDescriptor.cs

```diff
@@ -0,0 +1,62 @@
+//
+// NWAdvertiseDescriptor.cs: Bindings the Netowrk nw_advertise_descriptor_t API
+//
+// Authors:
+//   Miguel de Icaza (miguel@microsoft.com)
+//
+// Copyrigh 2018 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using nw_advertise_descriptor_t=System.IntPtr;
+using OS_nw_advertise_descriptor=System.IntPtr;
+
+namespace Network {
+	
+	[TV (12,0), Mac (10,14), iOS (12,0)]
+	public class NWAdvertiseDescriptor : NativeObject {
+		public NWAdvertiseDescriptor (IntPtr handle, bool owns) : base (handle, owns)
+		{ }
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr nw_advertise_descriptor_create_bonjour_service (string name, string type, string domain);
+
+		public static NWAdvertiseDescriptor CreateBonjourService (string name, string type = null, string domain = null)
```

Comment:
@migueldeicaza I would not add type to be optional, since that will most (if not always) result in the nw_advertise_descriptor_create_bonjour_service returning a null pointer. I added a comment on how to fix the issue in the tests.